native/src/main/java/io/ballerina/lib/data/csvdata/utils/CsvUtils.java

```diff
@@ -0,0 +1,383 @@
+
+        return headers;
+    }
+
+    public static Optional<Type> getMutableType(IntersectionType intersectionType) {
+        for (Type constituentType : intersectionType.getConstituentTypes()) {
+            if (constituentType.getTag() == TypeTags.READONLY_TAG) {
+                continue;
+            }
+            return Optional.of(constituentType);
+        }
+        return Optional.empty();
+    }
+
+    public static Object convertToBasicType(Object csv, Type targetType, CsvConfig config) {
+        if (csv == null) {
+            csv = config.nilValue;
+        }
+        return ValueUtils.convert(csv, targetType);
+    }
+
+    public static void checkRequiredFieldsAndLogError(Map<String, Field> filedHierarchy, boolean absentAsNilableType) {
+        filedHierarchy.values().forEach(field -> {
+            if (absentAsNilableType && field.getFieldType().isNilable()) {
+                return;
+            }
+            if (SymbolFlags.isFlagOn(field.getFlags(), SymbolFlags.REQUIRED)) {
+                throw DiagnosticLog.error(DiagnosticErrorCode.INVALID_FIELD_IN_CSV, field.getFieldName());
+            }
+        });
+    }
+
+    public static boolean isHeaderFieldsEmpty(Map<String, Field> currentField) {
+        for (Field field: currentField.values()) {
+            if (SymbolFlags.isFlagOn(field.getFlags(), SymbolFlags.REQUIRED)) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    public static boolean checkTypeCompatibility(Type constraintType, Object csv, boolean stringConversion) {
+        int tag = constraintType.getTag();
+        if (csv instanceof BString) {
+            if (stringConversion || tag == TypeTags.STRING_TAG
+                    || tag == TypeTags.CHAR_STRING_TAG || isJsonOrAnyDataOrAny(tag)) {
+                return true;
+            }
+        }
+        if (csv instanceof Long) {
+            if (tag == TypeTags.INT_TAG
+                    || tag == TypeTags.FLOAT_TAG || tag == TypeTags.DECIMAL_TAG || tag == TypeTags.BYTE_TAG
+                    || tag == TypeTags.SIGNED8_INT_TAG || tag == TypeTags.SIGNED16_INT_TAG
+                    || tag == TypeTags.SIGNED32_INT_TAG || tag == TypeTags.UNSIGNED8_INT_TAG
+                    || tag == TypeTags.UNSIGNED16_INT_TAG || tag == TypeTags.UNSIGNED32_INT_TAG
+                    || isJsonOrAnyDataOrAny(tag)) {
+                return true;
+            }
+        }
+        if (csv instanceof BDecimal) {
+            if ((tag == TypeTags.DECIMAL_TAG
+                    || tag == TypeTags.FLOAT_TAG || tag == TypeTags.INT_TAG) || isJsonOrAnyDataOrAny(tag)) {
+                return true;
+            }
+        }
+        if (csv instanceof Double) {
+            if ((tag == TypeTags.FLOAT_TAG
+                    || tag == TypeTags.DECIMAL_TAG || tag == TypeTags.INT_TAG) || isJsonOrAnyDataOrAny(tag)) {
+                return true;
+            }
+        }
+        if (Boolean.class.isInstance(csv)) {
```

Comment:
Isn't this same as `csv instanceOf Boolean`? I think  static version is better since that is the most commonly used one, even though this will most likely will get compiled to same instructions. WDYT?